packages/notte-core/src/notte_core/ast.py

```
@@ -673,8 +1236,28 @@ def get_safe_globals(self) -> dict[str, Any]:
 
         return restricted_globals
 
+    def safe_tmp_open(self, file: Any, mode: str = "r", *args: Any, **kwargs: Any):
+        """
+        Restricted open() replacement that only allows access under /tmp.
+        """
+        if kwargs.get("opener") is not None:
+            raise PermissionError("Custom openers are not allowed")
+
+        tmp_root = Path("/tmp").resolve()
```

Ruff (0.16.2)

[error] Probable insecure usage of temporary file or directory: "/tmp"

(S108)